### ISSUE TITLE
Open source options for all stages

### ISSUE THREAD

**Issue by bsridatta** [jacopotagliabue/you-dont-need-a-bigger-boat#12]
Thanks a lot for your work. Love your repos and podcasts. I wanted to suggest adding atleast one open source alternative for each stage. It would be helpful for folks that was to quickly build a full system with freely available open source tools. Thanks!

**Comment by jacopotagliabue** [comment]
hi @bsridatta ! Thanks for reaching out. Looking at the basic list, it seems most tools are already OS or available as OS. In particular:

- Metaflow is OS
- No data warehouse solution is OS as far as I know
- Prefect / Airflow are OS orchestrators
- dbt is OS
- Great Expectations is OS
- Experiment tracking seems the missing category: OS exists ([MlFlow](https://www.mlflow.org/docs/latest/index.html#)) but I find the additional cost of maintaining higher than the cost of SaaS such as [Comet](https://www.comet.ml/) or [Neptune](https://neptune.ai/).
- Model serving is similar: [Seldon](https://www.seldon.io/solutions/open-source-projects/core) is OS but it is way more complex than PaaS offerings.

Is there any specific tool / need not covered by the list?

**Comment by bsridatta** [comment]
Thanks for the quick reply @jacopotagliabue. 

You are right, most are OS indeed. I did not know Perfect was OS 😅 

Regarding experiment tracking:

- IMO MLFlow (OS version) is not hard to maintain (maybe I lack first hand experience) but is also not suitable for teams. 
- KubeFlow seems to be the only OS option that have experiment tracking that is usable for teams. But you cant just get that feature alone.
- So I guess only paid options for experiment tracking. I have only checked W&B and Comet and they seem to be too expensive, thanks for pointing out Neptune seems more reasonable.   

Thanks again!
